tap_kafka/sync.py

```diff
@@ -82,18 +84,38 @@ def init_kafka_consumer(kafka_config):
         # Non-configurable parameters
         'enable.auto.commit': False,
         'auto.offset.reset': 'earliest',
+        'value.deserializer': JSONSimpleDeserializer(),
     })
     consumer.subscribe([kafka_config['topic']])
 
     return consumer
 
 
+def get_timestamp_from_timestamp_tuple(kafka_ts):
+    """Get the actual timestamp value from a kafka timestamp tuple
+
+    It accepts tuple and list to be compatible with JSON serialised kafka messages as well
+    """
+    if isinstance(kafka_ts, (list, tuple)):
+        ts_type = kafka_ts[0]
+
+        if ts_type == confluent_kafka.TIMESTAMP_NOT_AVAILABLE:
+            return 0
```

Comment:
is this the right behavior? if the broker doesn't support  timestamps then bookmarking in state would not be possible.

Reply:
good question. Should we raise an exception?

Reply:
I added 0201e004a28ffcf91bacd4a5e9724b9e63e79f94 which makes timestamps mandatory. If timestamp not available in the message then we raise exception. You're right, there's no point of consuming messages without timestamps because we can't bookmark positions with it.

Reply:
is this something we can check in discovery mode and fail if it's broker doesn't support timestamps?

Reply:
no, this information comes only with the message itself.

Reply:
Oh, ok.